backend/handlers/createSession.js

```diff
@@ -0,0 +1,44 @@
+const { PrismaClient } = require("@prisma/client");
+const prisma = new PrismaClient();
+
+const createSession = ( async (request, response) => {
+    const { email, password } = request.body;
+
+    if(!email || !password) return response.status(400).send('Username and password are missing');
+
+    try {
+        const user = await prisma.user.findUnique({
+            where: {email}
+        });
+        if (!user) {
+            return response.status(404).json({message: "user not found"}) 
+        }
+        const passwordMatch = await bcrypt.compare(password, user.password)
+    
+        if (passwordMatch) {
+            request.session.user = user.id
+        }
+        else {
+            return response.status(400).send('invalid email or password')
+        }
```

Comment:
this is stylistic but typically we want to improve readability by reducing nesting.

```
if (!passwordMatch) {
  return response.status(400).send('invalid email or password')
}

request.session.user = user.id
```

reduces the nesting by one. it's a common pattern 

no change needed though